distributions/standalone/src/main/resources/bin/additional.groovy

```diff
@@ -159,3 +159,27 @@ def String getArchitecture() {
             return
     }
 }
+
+/**
+ *
+ * @param args
+ * @return the value of -additionalJavaProperties argument command line if It was set, DEFAULT_NUMBER_OF_ADDITIONAL_JAVA_ARGUMENTS otherwise.
+ */
+
+def int getNumberOfAdditionalJavaProperties(String[] args) {
+    int DEFAULT_NUMBER_OF_ADDITIONAL_JAVA_ARGUMENTS = 20;
+
+    //find the item in the list
+    def additionalJavaArguments = args.find { it ==~ /-additionalJavaProperties=(\d+)+/ }
+
+    //get the number of JavaArguments
+    additionalJavaArgumentsMatcher = additionalJavaArguments =~ /-additionalJavaProperties=(\d+)+/
+    additionalJavaArgumentsMatcher.find()
+
+    if(additionalJavaArgumentsMatcher.matches())
+    {
+        return additionalJavaArgumentsMatcher[0][1].toInteger() + 1
+    }
+
+    return DEFAULT_NUMBER_OF_ADDITIONAL_JAVA_ARGUMENTS + 1

```

Comment:
The approach of "to take advantage of the gaps in the numbering that Tanuki allows." works only on the latest versions of Tanuki wrapper. I'm not sure from which version that feature was added, so it won't work for old versions